src/main/java/edu/ucsb/cs156/happiercows/controllers/UserCommonsController.java

```diff
@@ -57,4 +69,59 @@ public UserCommons getUserCommonsById(
     return userCommons;
   }
 
-}
\ No newline at end of file
+    @ApiOperation(value = "Buy a cow, totalWealth updated")
+  @PreAuthorize("hasRole('ROLE_USER')")
+  @PutMapping("/buy")
+  public ResponseEntity<String> putUserCommonsByIdBuy(
+          @ApiParam("commonsId") @RequestParam Long commonsId, 
+            @RequestBody @Valid UserCommons incomingUserCommons) throws JsonProcessingException {
+        User u = getCurrentUser().getUser();
+        Long userId = u.getId();
+
+        Commons commons = commonsRepository.findById(commonsId).orElseThrow( 
+          ()->new EntityNotFoundException(Commons.class, commonsId));
+        UserCommons userCommons = userCommonsRepository.findByCommonsIdAndUserId(commonsId, userId)
+        .orElseThrow(
+            () -> new EntityNotFoundException(UserCommons.class, "commonsId", commonsId, "userId", userId));
```

Comment:
Instead of passing in a UserCommons object as the "incomingUserCommons", it would make more sense to just look up that using the commonsId, and the userId that you get from:
```
User u = getCurrentUser().getUser();
```

So when you do 

```
UserCommons userCommons = userCommonsRepository.findByCommonsIdAndUserId(commonsId, userId)
```

the userId would be u.getId():

And you'd remove the `@RequestBody @Valid UserCommons incomingUserCommons`